pkgs/development/python-modules/dgl/default.nix

```
@@ -0,0 +1,261 @@
+    "test_rpc_sampling_shuffle[1]"
+    "test_rpc_sampling_shuffle[2]"
+    "test_rpc_timeout[socket]"
+    "test_rpc_timeout[tensorpipe]"
+    "test_server_client"
+    "test_standalone"
+    "test_standalone_etype_sampling"
+    "test_standalone_sampling"
+  ];
+
+  disabledTestString = testsList: "not (${lib.strings.concatStringsSep " or " testsList})";
+in
+
+buildPythonPackage {
+  inherit pname version;
+
+  format = "other";
+
+  src = fetchFromGitHub {
+    owner = "dmlc";
+    repo = "dgl";
+    rev = "v${version}";
+    hash = "sha256-FQxfyHPvqI1x4b3byCrTAf11El2MNOXtb5/jx66Yve4=";
+    fetchSubmodules = true;
+  };
+
+  dependencies =
+    [
+      networkx
+      numpy
+      pandas
+      psutil
+      pydantic
+      requests
+      scikit-learn
+      scipy
+      tqdm
+    ]
+    ++ lib.optionals withTorch [ torch ]
+    ++ lib.optionals withTensorflow [
+      keras
+      tensorflow
+    ];
+
+  nativeCheckInputs = [ pytest ];
+
+  env = {
+    # error: 'uintptr_t' in namespace 'std' does not name a type
+    CXXFLAGS = "-include cstdint";
+
+    # error: ‘iadjwgt’ may be used uninitialized
+    NIX_CFLAGS_COMPILE = "-Wno-error=maybe-uninitialized";
+  };
+
+  cmakeFlags = [
+    "-DBUILD_CPP_TEST=ON"
+    "-DBUILD_SPARSE=OFF"
+    "-DUSE_LIBXSMM=OFF"
+  ];
+
+  nativeBuildInputs = [
+    cmake
+    cython
+    pip
+  ];
+
+  buildInputs = [
+    cudaPackages.cudatoolkit
+    gtest
+  ];
+
+  buildPhase = ''
+    runHook preBuild
+
+    make -j $NIX_BUILD_CORES
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out
+
+    pushd ../python
+    python3 -m pip install . --prefix=$out --no-index
+    python3 setup.py build_ext --inplace
+    popd
+
+    runHook postInstall
+  '';
+
+  doCheck = true;
+
+  checkPhase =
+    ''
+      runHook preCheck
+
+      export HOME="$(mktemp -d)"
+
+      cd ..
+
+      ./build/runUnitTests
+
+      export PYTHONPATH="$PYTHONPATH:$out/${python.sitePackages}:$src/tests"
+      export PATH="$PATH:$out/bin"
+    ''
+    + lib.optionalString withTorch ''
+      export DGLBACKEND=pytorch
+      pytest tests/python/pytorch/ -k '${disabledTestString pytorchDisabledTests}' \
+       --ignore=tests/python/pytorch/sparse/
+      pytest tests/python/common/ -k '${disabledTestString commonDisabledTests}'
+
+      pytest tests/distributed/ -k '${disabledTestString distributedDisabledTests}'
+    ''
+    +
+      lib.optionalString (withTensorflow && withTorch) # should be withTensorflow, but you need 'import torch' to run these
+        ''
+          export DGLBACKEND=tensorflow
+          pytest tests/python/tensorflow/
+          pytest tests/python/common/ -k '${disabledTestString commonDisabledTests}'
+        ''
+    + ''
+      runHook postCheck
+    '';
+
+  meta = with lib; {
+    description = ''
+      Python package built to ease deep learning on graph, on top of existing DL frameworks.
+    '';
+    homepage = "https://www.dgl.ai/";
```

```suggestion
    homepage = "https://www.dgl.ai/";
    changelog = "https://github.com/dmlc/dgl/releases/tag/${src.rev}"
```
Adding `meta.changelog` should be useful for future updates.